corehq/apps/case_importer/suggested_fields.py

```diff
@@ -33,11 +33,14 @@ def _combine_field_specs(field_specs, exclude_fields):
 def get_suggested_case_fields(domain, case_type, exclude=None):
     exclude_fields = set(RESERVED_FIELDS) | set(exclude or [])
     hints_dict = get_values_hints_dict(domain, case_type)
+    deprecated_fields = get_deprecated_fields(domain, case_type)
```

Comment:
I think this should check the DATA_DICTIONARY flag is on (either here or in the function).

Reply:
Hmm, should the `get_values_hints_dict` also be suppressed if data dictonary is not enabled? For that matter should the whole column for "Valid values and formats" be suppressed when this flag is not on?

Reply:
Good point. Yeah, the page shouldn't show the column if the feature isn't turned on. Turning it off at the UI level then probably makes it not really matter if `get_values_hints_dict` and `deprecated_fields` are populated here.